src/lib/remark-plugins/rewrite-tutorial-links/index.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * This plugin writes urls for learn tutorial content that reference other
+ * learn tutorials or collections.
+ *
+ * If the link references a tutorial or collection that is included in the
+ * beta-product list, this rewrites the link path to be relative to dev dot.
+ *
+ * All other tutorial and collection links are rewritten to point externally
+ * to production learn.hashicorp.com
+ *
+ * ONCE DEV PORTAL IS GENERALLY AVAILBLE, this plugin will be deprecated
+ * as all written internal links within the content will be updated in the
+ * content itself. This is an interim workaround while we have a divergent state
+ * during beta.
+ */
+
+import { Link, Definition } from 'mdast'
+import { Plugin } from 'unified'
+import { visit } from 'unist-util-visit'
+//import moize, { Options } from 'moize'
+import { ProductOption } from 'lib/learn-client/types'
+import {
+  getTutorialMap,
+  handleCollectionLink,
+  handleTutorialLink,
+} from './utils'
+
+const learnProducts = new RegExp(Object.keys(ProductOption).join('|'), 'g')
+const learnLink = new RegExp('(learn.hashicorp.com|collections|tutorials)')
+let TUTORIAL_MAP
+
+export const rewriteTutorialLinksPlugin: Plugin = () => {
+  return async function transformer(tree) {
+    TUTORIAL_MAP = await getTutorialMap()
+
+    visit(tree, 'link', handleRewriteTutorialsLink)
+    visit(tree, 'definition', handleRewriteTutorialsLink)
+  }
+}
+
+function handleRewriteTutorialsLink(node: Link | Definition) {
+  try {
+    // return early if non tutorial or collection link
+    if (!learnLink.test(node.url)) {
+      return
+    }
+
+    const [product] = node.url.match(learnProducts)
+    const isExternalLearnLink = node.url.includes('learn.hashicorp.com')
+    const isBetaProduct = __config.dev_dot.beta_product_slugs.includes(product)
```

Comment:
nit, consider using `getIsBetaProduct` from `src/lib/get-is-beta-product.ts`

Reply:
Wonderful, will do